sdk/client/commander.go

```diff
@@ -314,27 +315,28 @@ func (c *commander) createClient() error {
 func (c *commander) recvLoop() {
 	log.Debug("Commander receive loop starting")
 	for {
-		err := sdk.WaitUntil(c.ctx, c.backoffSettings.initialInterval, c.backoffSettings.maxInterval, c.backoffSettings.maxTimeout, func() error {
-			cmd, err := c.channel.Recv()
-			log.Infof("Commander received %v, %v", cmd, err)
-			if err != nil {
-				return c.handleGrpcError("Commander Channel Recv", err)
-			}
+		err := sdk.WaitUntilWithJitterAndMultiplier(c.ctx, c.backoffSettings.initialInterval, c.backoffSettings.maxInterval,
+			c.backoffSettings.maxTimeout, c.backoffSettings.randomization_factor, c.backoffSettings.multiplier, func() error {
+				cmd, err := c.channel.Recv()
+				log.Infof("Commander received %v, %v", cmd, err)
+				if err != nil {
```

Comment:
Would it be better to update the commander.go plugin to process `Command_AgentConfig` messages and call `c.cmdr.WithBackoffSettings(....)` to update the backoff settings?

https://github.com/nginx/agent/blob/main/src/plugins/commander.go#L68

Reply:
I initially had the same approach, but it seems `BackoffSettings` struct fields so defined in `client.go` https://github.com/nginx/agent/blob/main/sdk/client/client.go#L22 is un-exported in nature.

If we want to go via this approach, we have to make them exported. Is that route ok with you?